src/pages/index.js

```diff
@@ -0,0 +1,293 @@
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = document.querySelector("#delete-form");
+const cardForm = document.querySelector("#card-form");
+const avatarForm = document.querySelector("#edit-avatar-form");
+const editProfileForm = document.querySelector("#edit-profile-form");
+const previewModal = document.querySelector("#preview-modal");
+
+// Open and close modal functions
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscClose);
+  modal.addEventListener("mousedown", handleOverlayClick);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscClose);
+  modal.removeEventListener("mousedown", handleOverlayClick);
+}
+
+// Escape key close handler
+function handleEscClose(event) {
+  if (event.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    if (openedModal) closeModal(openedModal);
+  }
+}
+
+// Overlay click close handler
+function handleOverlayClick(event) {
+  if (event.target.classList.contains("modal_opened")) {
+    closeModal(event.target);
+  }
+}
+
+// Helper to set up modal close buttons
+function setUpCloseButtons() {
+  document.querySelectorAll(".modal__close-btn").forEach((button) => {
+    const modal = button.closest(".modal");
+    button.addEventListener("click", () => closeModal(modal));
+  });
+}
+setUpCloseButtons();
+
+// Load user info and cards
+api
+  .getAppInfo()
+  .then(([cards, userInfo]) => {
+    document.querySelector(".profile__avatar").src = userInfo.avatar;
+    document.querySelector(".profile__name").textContent = userInfo.name;
+    document.querySelector(".profile__description").textContent =
+      userInfo.about;
+
+    const cardsList = document.querySelector(".cards__list");
+    cards
+      .reverse()
+      .forEach((cardData) => cardsList.prepend(createCardElement(cardData)));
+  })
+  .catch(console.error);
+
+// Card creation function
+function createCardElement(data) {
+  const template = document.querySelector("#card-template");
+  const cardElement = template.content.querySelector(".card").cloneNode(true);
+
+  const cardImageEl = cardElement.querySelector(".card__image");
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  cardElement.querySelector(".card__title").textContent = data.name;
+
+  setUpImagePreview(cardImageEl, data);
+  setUpLikeButton(cardElement.querySelector(".card__like-btn"), data);
+  cardElement
+    .querySelector(".card__delete-btn")
+    .addEventListener("click", () => openDeleteModal(cardElement, data._id));
+
+  return cardElement;
+}
+
+// Helper function to set up image preview
+function setUpImagePreview(imageElement, data) {
+  imageElement.addEventListener("click", () => {
+    const modalImage = previewModal.querySelector(".modal__image");
+    const modalCaption = previewModal.querySelector(".modal__caption");
```

Comment:
* It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine